dockerproxy/dockerd.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/mitchellh/go-ps"
+	"github.com/pkg/errors"
+)
+
+func runDockerd() (func() error, *client.Client, error) {
+	// noop
+	if noDockerd {
+		client, err := client.NewClientWithOpts(client.FromEnv)
+		if err != nil {
+			return nil, nil, err
+		}
+		return func() error { return nil }, client, nil
+	}
+
+	// just to be sure, because machines now reuse snapshots
+	os.RemoveAll("/var/run/docker.pid")
+
+	// Launch `dockerd`
+	dockerd := exec.Command("dockerd", "-p", "/var/run/docker.pid")
+	dockerd.Stdout = os.Stderr
+	dockerd.Stderr = os.Stderr
+
+	if err := dockerd.Start(); err != nil {
+		return nil, nil, errors.Wrap(err, "could not start dockerd")
+	}
+
+	cmd := exec.Command("docker", "buildx", "inspect", "--bootstrap")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		log.Warnln("Error bootstrapping buildx builder:", err)
+	}
+
+	dockerDone := make(chan struct{})
+
+	go func() {
+		if err := dockerd.Wait(); err != nil {
+			log.Errorf("error waiting on docker: %v", err)
+		}
+		close(dockerDone)
+	}()
+
+	healthCtx, healthCancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer healthCancel()
```

Comment:
Move the `10*time.Second` to a constant so you can use it in the `fmt.Errorf()` below.